src/coreclr/nativeaot/Runtime/riscv64/WriteBarriers.S

```diff
@@ -295,12 +306,12 @@ LEAF_ENTRY RhpCheckedLockCmpXchg
 
 LOCAL_LABEL(CmpXchgRetry):
         // Load the current value at the destination address.
-        lr.d    t0, (a0)         // t0 = *dest
```

Comment:
Does `RhpAssignRefRiscV64` need a `fence` too?

Reply:
If I were to go with ARM as example we should probably do `fence r, rw` there as well.

https://github.com/dotnet/runtime/blob/5f97ddfff8074355964e6d13a690c236dd818050/src/coreclr/nativeaot/Runtime/arm/WriteBarriers.S#L154-L159

Reply:
Fixed. I compared the fences with the CoreCLR code. It has one additional `fence rw, rw` in the shadow heap code which should not be necessary:
https://github.com/dotnet/runtime/blob/9ccb9dee75d4cb54897bd874b8e1c555c4f5d779/src/coreclr/vm/riscv64/asmhelpers.S#L229-L231